parser.py

```diff
@@ -84,8 +89,11 @@ def parse_str_table(data: str) -> Table:
 
     """
     table = Table()
-    unformatted_header, unformatted_body = split_str_table(data)
-    table.fields = extract_values_from_row(unformatted_header)
+    if table_with_header:
+        unformatted_header, unformatted_body = split_str_table(data)
+        table.fields = extract_values_from_row(unformatted_header)
+    else:
+        no_header, unformatted_body = split_str_table(f'no_header\n{data}')
```

Comment:
Maybe just split the lines here, instead of creating the fake header line?

My preference ^^